sda-commons-server-openapi/src/main/java/org/sdase/commons/server/openapi/filter/OpenAPISpecFilterSet.java

```diff
@@ -0,0 +1,198 @@
+package org.sdase.commons.server.openapi.filter;
+
+import io.swagger.v3.core.filter.OpenAPISpecFilter;
+import io.swagger.v3.core.model.ApiDescription;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.Operation;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.parameters.Parameter;
+import io.swagger.v3.oas.models.parameters.RequestBody;
+import io.swagger.v3.oas.models.responses.ApiResponse;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * An implementation of the {@link OpenAPISpecFilter} that delegates the request to multiple
+ * registered filters. Since swagger-core always creates a new instance of this filter, the
+ * registered filters are stored statically.
+ *
+ * <p>These filters is applied on every call to /openapi.{yaml|json} and can inject information from
+ * the request context.
+ */
+public class OpenAPISpecFilterSet implements OpenAPISpecFilter {
+  private static final List<OpenAPISpecFilter> filters = new ArrayList<>();
+
+  /**
+   * Register a filter that should be called when an OpenAPI file is returned to a caller.
+   *
+   * @see #clear()
+   * @param filter the filter to register
+   */
+  public static void register(OpenAPISpecFilter filter) {
+    filters.add(filter);
+  }
+
+  /** Clear all filters */
+  public static void clear() {
+    filters.clear();
+  }
+
+  @Override
+  public Optional<OpenAPI> filterOpenAPI(
+      OpenAPI openAPI,
+      Map<String, List<String>> params,
+      Map<String, String> cookies,
+      Map<String, List<String>> headers) {
+
+    for (OpenAPISpecFilter filter : filters) {
+      Optional<OpenAPI> openAPI1 = filter.filterOpenAPI(openAPI, params, cookies, headers);
+      if (openAPI1.isPresent()) {
+        openAPI = openAPI1.get();
+      }
+    }
+
+    return Optional.of(openAPI);
+  }
```

Comment:
I think it is more readable to use the optional API in such cases. One may also skip introducing the new `result` variable and use the openAPI from the signature. But that is usually not good practice.

```suggestion
    OpenAPI result = openAPI;
    for (OpenAPISpecFilter filter : filters) {
      result = filter.filterOpenAPI(result, params, cookies, headers).orElse(result);
    }

    return Optional.ofNullable(result);
  }
```
I wonder what the use case of an empty `Optional` is in this API. I can't find any documentation in the interface.

Reply:
I came up with a shorter and cleaner implementation of this class.

Regarding the non-present Optional: This can be used to filter the OpenAPI definition, i.e. to remove certain properties, schemas, etc. just before returning the spec to the requester.